build/libllvm.ps1

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+. (Join-Path $PSScriptRoot "utils.ps1")
+
+Assert (Test-Path env:\PKG_NAME) "PKG_NAME was not defined"
+Assert (Test-Path env:\LINUX_PKG_NAME) "LINUX_PKG_NAME was not defined"
+Assert (Test-Path env:\WINDOWS_PKG_NAME) "WINDOWS_PKG_NAME was not defined"
+Assert (Test-Path env:\DARWIN_PKG_NAME) "DARWIN_PKG_NAME was not defined"
+Assert (Test-Path env:\AQ_LLVM_PACKAGE_GIT_VERSION) "AQ_LLVM_PACKAGE_GIT_VERSION was not defined"
+Assert (Test-Path env:\BLOBS_OUTDIR) "BLOBS_OUTDIR was not defined"
+Assert (Test-Path env:\BUILD_ARTIFACTSTAGINGDIRECTORY) "BUILD_ARTIFACTSTAGINGDIRECTORY was not defined"
+
+$llvmArchiveFiles = @(
+    "$($env:LINUX_PKG_NAME)/external/llvm-project/build/$($env:LINUX_PKG_NAME).tar.gz",
+    "$($env:WINDOWS_PKG_NAME)/external/llvm-project/build/$($env:WINDOWS_PKG_NAME).zip",
+    "$($env:DARWIN_PKG_NAME)/external/llvm-project/build/$($env:DARWIN_PKG_NAME).tar.gz"
+)
+
+for ($i = 0; $i -lt $llvmArchiveFiles.Count; $i++) {
```

Comment:
Out of curiousity, what motivates a `for` loop here versus `foreach` or `ForEach-Object`?

Reply:
I wanted to save the fully resolved path back into the original collection. Stylistic choice, I didn't like the alternative
```pwsh
$llvmArchiveFiles = $llvmArchiveFiles | ForEach-Object {
    Write-Vso "Raw: $_"
    $path = Join-Path (Get-Location) '..' $_
    Write-Vso "Joined: $path"
    $resolvedPath = Resolve-Path $path
    Write-Vso "Resolved: $resolvedPath"
    $resolvedPath
}
```